File: app/src/main/java/com/suishi/live/app/ui/activity/PortraitActivity.java
package com.suishi.live.app.ui.activity;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;

import com.suishi.live.app.R;
import com.suishi.live.app.widgets.MultiToggleImageButton;
import com.suishi.sslive.ui.LiveManager;
import com.suishi.sslive.widgets.CameraGlSurfaceView;

public class PortraitActivity extends Activity {

    CameraGlSurfaceView filter;

    private LiveManager mLiveManager;

    private MultiToggleImageButton mMicBtn;

    private MultiToggleImageButton mFlashBtn;

    private MultiToggleImageButton mFaceBtn;

    private MultiToggleImageButton mBeautyBtn;

    private MultiToggleImageButton mFocusBtn;

    private ImageButton mRecordBtn;

    private boolean isGray;

    private boolean isRecording;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_portrait);
        filter = findViewById(R.id.surface);
        mLiveManager = new LiveManager(this, filter);
        initViews();
        initListeners();
    }

    /**
     *
     */
    @Override
    protected void onResume() {
        super.onResume();

    }

    /**
     * Dispatch onPause() to fragments.
     */
    @Override
    protected void onPause() {
        super.onPause();
        mLiveManager.onStopStream();
    }

    private void initViews() {
        //  mLFLiveView = (LivingView) findViewById(R.id.liveView);
        mMicBtn = findViewById(R.id.record_mic_button);
        mFlashBtn = findViewById(R.id.camera_flash_button);
        mFaceBtn = findViewById(R.id.camera_switch_button);
        mBeautyBtn = findViewById(R.id.camera_render_button);
        mFocusBtn = findViewById(R.id.camera_focus_button);
        mRecordBtn = findViewById(R.id.btnRecord);
    }

    private void initListeners() {
        mMicBtn.setOnStateChangeListener((view, state) -> {
            //   mLFLiveView.mute(true);
        });
        mFlashBtn.setOnStateChangeListener((view, state) -> {
            //  mLFLiveView.switchTorch();
        });
        mFaceBtn.setOnStateChangeListener((view, state) -> {
            // 切换摄像头
            mLiveManager.switchCamera();
        });

        mBeautyBtn.setOnStateChangeListener((view, state) -> isGray = !isGray);

        mFocusBtn.setOnStateChangeListener((view, state) -> {
            // mLFLiveView.switchFocusMode();
        });

        mRecordBtn.setOnClickListener(v -> {
            if (isRecording) {
                //停止推流
                mRecordBtn.setBackgroundResource(R.mipmap.ic_record_start);
                mLiveManager.onStopStream();
                isRecording = false;
            } else {
                //开始推流
                mRecordBtn.setBackgroundResource(R.mipmap.ic_record_stop);
                mLiveManager.onStartStream();
                isRecording = true;
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/suishi/live/app/ui/activity/PortraitActivity.kt
package com.suishi.live.app.ui.activity

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ImageButton
import com.suishi.live.app.R
import com.suishi.live.app.widgets.MultiToggleImageButton
import com.suishi.live.app.widgets.MultiToggleImageButton.OnStateChangeListener
import com.suishi.sslive.ui.LiveManager

/**
 * 推流
 */
class PortraitActivity : Activity() {

    companion object{
        fun startActivity(context: Context){
            val intent = Intent(context, PortraitActivity::class.java)
            context.startActivity(intent)
        }
    }

    /**
     *
     */
   // var filter: CameraGlSurfaceView? = null
    /**
     *
     */
    private var mLiveManager: LiveManager? = null
    /**
     *
     */
    private var mMicBtn: MultiToggleImageButton? = null
    /**
     *
     */
    private var mFlashBtn: MultiToggleImageButton? = null
    /**
     *
     */
    private var mFaceBtn: MultiToggleImageButton? = null
    /**
     *
     */
    private var mBeautyBtn: MultiToggleImageButton? = null
    /**
     *
     */
    private var mFocusBtn: MultiToggleImageButton? = null
    /**
     *
     */
    private var mRecordBtn: ImageButton? = null
    /**
     *
     */
    private var isGray = false
    /**
     *
     */
    private var isRecording = false

    /**
     *
     */
    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_portrait)

        mMicBtn = (findViewById(R.id.record_mic_button) as MultiToggleImageButton).apply{
            setOnStateChangeListener(object :OnStateChangeListener{
                override fun stateChanged(view: View?, state: Int) {

                }

            })
        }
        mFlashBtn = (findViewById(R.id.camera_flash_button) as MultiToggleImageButton).apply{
            setOnStateChangeListener(object : OnStateChangeListener{
                override fun stateChanged(view: View?, state: Int) {

                }
            })
        }
        mFaceBtn = (findViewById(R.id.camera_switch_button)  as MultiToggleImageButton).apply{
            setOnStateChangeListener(object :OnStateChangeListener{
                override fun stateChanged(view: View?, state: Int) {
                    // 切换摄像头
                   // mLiveManager!!.switchCamera()
                }

            })
        }
        mBeautyBtn = (findViewById(R.id.camera_render_button) as MultiToggleImageButton).apply{
            setOnStateChangeListener(object :OnStateChangeListener{
                override fun stateChanged(view: View?, state: Int) {
                    isGray = !isGray
                }

            })
        }
        mFocusBtn = (findViewById(R.id.camera_focus_button)   as MultiToggleImageButton).apply{
            setOnStateChangeListener(object :OnStateChangeListener{
                override fun stateChanged(view: View?, state: Int) {

                }

            })
        }
        mRecordBtn = (findViewById(R.id.btnRecord) as ImageButton).apply{
            setOnClickListener{ v: View? ->
                isRecording = if (isRecording) {
                    //停止推流
                    mRecordBtn!!.setBackgroundResource(R.mipmap.ic_record_start)
                    mLiveManager!!.onStopStream()
                    false
                } else {
                    //开始推流
                    mRecordBtn!!.setBackgroundResource(R.mipmap.ic_record_stop)
                    mLiveManager!!.onStartStream()
                    true
                }
            }
        }
    }

    /**
     * Dispatch onPause() to fragments.
     */
    override fun onPause() {
        super.onPause()
        mLiveManager!!.onStopStream()
    }


    override fun onDestroy() {
        super.onDestroy()
    }
}